crates/rome_js_formatter/src/js/expressions/static_member_expression.rs

```diff
@@ -12,10 +15,35 @@ impl FormatNode for JsStaticMemberExpression {
             member,
         } = self.as_fields();
 
-        Ok(group_elements(format_elements![
-            object.format(formatter)?,
-            operator_token.format(formatter)?,
-            member.format(formatter)?,
-        ]))
+        let object_syntax = object.clone()?.syntax().clone();
+
+        let is_object_number_literal =
+            object_syntax.kind() == JsSyntaxKind::JS_NUMBER_LITERAL_EXPRESSION;
+
+        let has_object_trailing_whitespace = object_syntax
+            .text()
+            .to_string()
+            .chars()
+            .last()
+            .unwrap()
+            .is_whitespace();
+
+        let formatted_object = object?.format(formatter)?;
+
+        if is_object_number_literal && has_object_trailing_whitespace {
+            Ok(group_elements(format_elements![
+                token("("),
+                formatted_object,
+                token(")"),
```

Comment:
I would suggest to use `hard_group_elements` here, so possible comments are kicked outside of the parenthesis.

```suggestion
				hard_group_elements(format_elements![
        	        token("("),
    	            formatted_object,
	                token(")"),				
				]),
```